src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -128,12 +131,81 @@ public DefinitionRegistrationOptions GetRegistrationOptions(DefinitionCapability
                 new LocationOrLocationLink(new Location
                 {
                     Uri = originComponentUri,
-                    Range = new Range(new Position(0, 0), new Position(0, 0)),
+                    Range = range,
                 }),
             });
         }
 
-        internal static async Task<TagHelperBinding?> GetOriginTagHelperBindingAsync(
+        private async Task<Range> GetNavigateRangeAsync(DocumentSnapshot documentSnapshot, BoundAttributeDescriptor? attributeDescriptor, CancellationToken cancellationToken)
+        {
+            if (attributeDescriptor is not null)
+            {
+                var originCodeDocument = await documentSnapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
+                var range = await TryGetPropertyRangeAsync(originCodeDocument, attributeDescriptor.GetPropertyName(), _documentMappingService, cancellationToken).ConfigureAwait(false);
+
+                if (range is not null)
+                {
+                    return range;
+                }
+            }
+
+            return new Range(new Position(0, 0), new Position(0, 0));
+        }
+
+        internal static async Task<Range?> TryGetPropertyRangeAsync(RazorCodeDocument codeDocument, string propertyName, RazorDocumentMappingService documentMappingService, CancellationToken cancellationToken)
+        {
+            // Parse the C# file and find the property that matches the name, and that has a [Parameter] attribute
+
+            var csharpText = codeDocument.GetCSharpSourceText();
+            var syntaxTree = CSharpSyntaxTree.ParseText(csharpText, cancellationToken: cancellationToken);
+            var root = await syntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var property = root.DescendantNodes(n => n is BaseNamespaceDeclarationSyntax or CompilationUnitSyntax or ClassDeclarationSyntax)
```

Comment:
Need to be smarter about this. Add a test with:

```
@code {
   class C
   {
      string Title { get; }
   }

   string Title { get; }
}
```

It will find the inner prop, which is wrong.

Reply:
ooo good call